suricata/update/parsers.py

```diff
@@ -0,0 +1,250 @@
+    global_parser = argparse.ArgumentParser(add_help=False)
+
+    for arg, opts in arguments:
+        global_parser.add_argument(*arg, **opts)
+
+    return global_parser
+
+
+def parse_update(subparsers, global_parser, arguments):
+    # The "update" (default) sub-command parser.
+    update_parser = subparsers.add_parser(
+        "update", add_help=True, parents=[global_parser],
+        formatter_class=argparse.RawDescriptionHelpFormatter)
+
+    for arg, opts in arguments:
+        update_parser.add_argument(*arg, **opts)
+
+    return update_parser
+
+
+def parse_commands(subparsers, global_parser):
+    commands.listsources.register(subparsers.add_parser(
+        "list-sources", parents=[global_parser]))
+    commands.listenabledsources.register(subparsers.add_parser(
+        "list-enabled-sources", parents=[global_parser]))
+    commands.addsource.register(subparsers.add_parser(
+        "add-source", parents=[global_parser]))
+    commands.updatesources.register(subparsers.add_parser(
+        "update-sources", parents=[global_parser]))
+    commands.enablesource.register(subparsers.add_parser(
+        "enable-source", parents=[global_parser]))
+    commands.disablesource.register(subparsers.add_parser(
+        "disable-source", parents=[global_parser]))
+    commands.removesource.register(subparsers.add_parser(
+        "remove-source", parents=[global_parser]))
+
+
+def parse_arg():
+    default_update_yaml = config.DEFAULT_UPDATE_YAML_PATH
+
+    arguments = [
+        (("-v", "--verbose"),
+            {'action': 'store_true', 'default': None,
+                'help': "Be more verbose"}),
+        (("-q", "--quiet"),
+            {'action': 'store_true', 'default': None,
+                'help': "Be quiet, warning and error messages only"}),
+        (("-D", "--data-dir"),
+            {'metavar': '<directory>', 'dest': 'data_dir',
+                'help': "Data directory (default: /var/lib/suricata)"}),
+        (("-c", "--config"),
+            {'metavar': '<filename>',
+                'help': "configuration file (default: %s)"
+                % (default_update_yaml)}),
+        (("--suricata-conf",),
+            {'metavar': '<filename>',
+                'help': "configuration file "
+                "(default: /etc/suricata/suricata.yaml)"}),
+        (("--suricata",),
+            {'metavar': '<path>',
+                'help': "Path to Suricata program"}),
+        (("--suricata-version",),
+            {'metavar': '<version>',
+                'help': "Override Suricata version"}),
+        (("--user-agent",),
+            {'metavar': '<user-agent>',
+                'help': "Set custom user-agent string"}),
+        (("--no-check-certificate",),
+            {'action': 'store_true', 'default': None,
+                'help': "Disable server SSL/TLS certificate verification"}),
+        (("-V", "--version"),
+            {'action': 'store_true', 'default': False,
+                'help': "Display version"})
+    ]
```

Comment:
Sorry for not being clear the last time. I was suggesting moving this argument list outside the function. It would make the function more compact and not all about declarations.

Reply:
@shivan1b I did move the argument list out of the `parse_global` and `parse_update` functions. It's unclear which function here are you referring to? And where should I move these lists?